### ISSUE TITLE
Bulk upload possible with duplicate roll number

### ISSUE THREAD

**Issue by madhumitadey** [OUPMATH/OxfordMath_UAT#109]
On trying to upload bulk excel, we are able to upload students with duplicate roll number. Please see the screenshot below. Is this okay?

![image](https://cloud.githubusercontent.com/assets/16831474/13455296/7149f2c6-e082-11e5-8b61-54c787bc353d.png)

**Comment by manasghosh81** [comment]
@madhumitadey , @mayank2k2 : The bulk upload is assumed to be done on one-shot upload for the school. This case has occurred since probably you have uploaded students twice for the same school (not a promotion case though). If class, roll number are same in the single excel sheet, it will error out. In this case, student was already existing and then you uploaded a different student with the same roll number. 
If this is to be handled, it has to be taken up as an enhancement during  the support time.

Thanks
Subha

**Comment by sumitmago** [comment]
Dear Subha,
               It is definitely a natural business case that we can’t have two same role numbers in same class. So I will not debate on whether it is a miss or enhancement but let’s close this even if we have to do it in next couple of days.

We definitely discussed that even in the case of second upload in second tranche, validations must be in place for checking duplicity (which may happen because of human error/mistake’s).

Thanks & Regards
Sumit Mago
Head (Digital Solutions) – Learning & Assessment
Oxford University Press
2nd Floor, Tower-1, Stellar Crest  Business Center, IT Park, C-25 Sector 62,  Noida
(D) - 0120-6718900 Extn - 214, (M) +91-9873299973

The ABC of Innovation:
Conceive
Believe
Achieve

**Comment by manasghosh81** [comment]
@sumitmago : Hi Sumit,

You are getting it wrong. We discussed during our FSD, that by mistake user may upload the same excel again, in that case it should be restricted  and we have restricted it accordingly in a way that if first name, middle name, last name, roll number, class and section are same it will not allow the upload. 

Now you are saying that we need to restrict if roll number, class and section are same. So will remove the first name, middle name, last name from the checking and do it.

Pls let me know if this would work. 

It would have been better if you would have given us a list of all validations instead of we doing the validations and request for changes coming up 2 days before the go-live.

Subha.
